File: 백준/Silver/1966. 프린터 큐/프린터 큐.py
import sys
from collections import deque
input = sys.stdin.readline

T = int(input())

for _ in range(T):
    n,m =map(int, input().split())
    que = deque(map(int, input().split()))
    idx = deque(range(0,n))
    answer=0
    
    while True:
        
        # 대기열 가장 앞에 있는 것 = 가장 중요도 높은것
        if que[0]==max(que):
            answer +=1
            
            # 동일한 중요도이면서 찾던 문서 번호임
            if idx[0]==m:
                print(answer)
                break
            else:
                que.popleft()
                idx.popleft()
        
        # 대기열 가장 앞에 있는것 != 가장 중요도 높은 것
        # 대기열 뒤로 넘기기
        else:
            que.append(que.popleft())
            idx.append(idx.popleft())

File: 백준/Silver/1966. 프린터 큐/프린터 큐.js
let fs = require("fs")
let line = fs.readFileSync("/dev/stdin").toString().split("\n");

let T = Number(line[0])


function solution(arr, m){

  count=0
  while(arr.length!=0){
    let max_ = arr.reduce((a, b) => (a[0] >= b[0] ? a : b));
      let v = arr.shift();
      //console.log(max_,v,arr)
      if (v!=max_){
        arr.push(v)
      }
      else{
        count+=1
        if(v[1]==m){
          break;
        }
      }
    }
      return count;
  }






for(let i = 1;i<=T*2;i+=2){
let [n, m] = line[i].split(" ").map((x) => Number(x));
let arr = line[i+1].split(" ").map((x,idx) => [Number(x),idx]);
//console.log(n,m,arr)
console.log(solution(arr, m))
}




